examples/diffusion/python_stable_diffusion_xl/txt2img.py

```diff
@@ -246,11 +246,17 @@ def __init__(self, onnx_model_path, compiled_model_path,
         elif "all" in fp16:
             fp16 = ["vae", "clip", "clip2", "unetxl"]
 
+        self.vae_name = "vae_decoder"
+        self.vae_input_name = "latent_sample"
+        if "vae" in fp16:
+            self.vae_name = "vae_fp16_fix"
+            self.vae_input_name = "input.1"
+
         print("Load models...")
         self.models = {
             "vae":
             StableDiffusionMGX.load_mgx_model(
-                "vae_decoder", {"latent_sample": [1, 4, 128, 128]},
+                self.vae_name, {self.vae_input_name: [1, 4, 128, 128]},
```

Comment:
Do you need to update steps in README.md about where to get this model ?

Reply:
@umangyadav can you point to where to get the model so the README can be adjusted

Reply:
https://huggingface.co/madebyollin/sdxl-vae-fp16-fix

Reply:
Can you try to export it like this and check if the result is correct?

```python
import torch
from diffusers import AutoencoderKL
import os

class VAEDecoder(torch.nn.Module):
    def __init__(self, vae):
        super().__init__()
        self.vae = vae

    def forward(self, latent_sample):
        return self.vae.decode(latent_sample)

vae = AutoencoderKL.from_pretrained("madebyollin/sdxl-vae-fp16-fix")
vae.eval()
output = "models/sdxl-1.0-base/vae_decoder_fp16_fix/model.onnx"
os.makedirs(os.path.dirname(output), exist_ok=True)
torch.onnx.export(VAEDecoder(vae),
                  torch.randn(1, 4, 128, 128),
                  output,
                  export_params=True,
                  do_constant_folding=True,
                  input_names=['latent_sample'])
```

If it works, it could be added next to `clip_modifier.py` like `export_vae_fp16_fix.py` or similar, and add a note in the README to use it before running `--fp16 vae`. Also, since the input-name can be set, only `vae_name` needed.
```python
vae_name = "vae_decoder_fp16_fix" if "vae" in fp16 else "vae_decoder"
```

Reply:
You can also cherry-pick [this](https://github.com/ROCm/AMDMIGraphX/commit/a605f925851e76b63cd9b25210c42f11bd143bdd).

Reply:
I made slightly different modifications, but used your above suggestion. Please take a look.